File: scratch/scratch_search.py
import os
import re

workspace = r"g:\Shebafi\sheba 22 2nd round"
logic_path = os.path.join(workspace, "controllers", "logic.php")

with open(logic_path, "r", encoding="utf-8", errors="ignore") as f:
    lines = f.readlines()

print(f"Total lines in logic.php: {len(lines)}")

# Let's search for "recharge" case insensitive
recharge_matches = []
for idx, line in enumerate(lines):
    if "recharge" in line.lower() or "left" in line.lower() or "active" in line.lower():
        # Filter matching lines
        if "recharge" in line.lower() or "status" in line.lower():
            recharge_matches.append((idx + 1, line.strip()))

print(f"Found {len(recharge_matches)} matches for 'recharge' or 'status'. Let's show first 100:")
for num, content in recharge_matches[:100]:
    print(f"L{num}: {content[:120]}")

File: scratch/scratch_search.php
<?php
$workspace = "g:/Shebafi/sheba 22 2nd round";
$logic_path = "$workspace/controllers/logic.php";

if (!file_exists($logic_path)) {
    die("logic.php does not exist at $logic_path\n");
}

$lines = file($logic_path);
echo "Total lines: " . count($lines) . "\n";

$recharge_matches = [];
foreach ($lines as $idx => $line) {
    $line_lower = strtolower($line);
    if (strpos($line_lower, 'recharge') !== false || strpos($line_lower, 'status') !== false || strpos($line_lower, 'mikrotik') !== false) {
        $recharge_matches[] = [
            'line' => $idx + 1,
            'content' => trim($line)
        ];
    }
}

echo "Found " . count($recharge_matches) . " matches.\n";
foreach (array_slice($recharge_matches, 0, 150) as $match) {
    echo "L" . $match['line'] . ": " . substr($match['content'], 0, 150) . "\n";
}
